sentry/src/main/java/io/sentry/config/CompositePropertiesProvider.java

```diff
@@ -31,12 +31,10 @@ public CompositePropertiesProvider(@NotNull List<PropertiesProvider> providers)
 
   @Override
   public @NotNull Map<String, String> getMap(final @NotNull String property) {
+    final Map<String, String> result = new ConcurrentHashMap<>();
     for (final PropertiesProvider provider : providers) {
```

Comment:
// can we add a comment about the precedence over providers? like:

// system property
// env vars.
// simple properties
// ...

just so that its easier to debug when tags are either missing or overwritten

Reply:
The order of providers is defined in `PropertiesProviderFactory` where it's documented. `CompositePropertiesProvider` technically is unaware of the specific order.

Reply:
indeed, sounds good then, thanks